src/AuthModule/components/Login/Login.tsx

```diff
@@ -1,10 +1,142 @@
-
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { useForm } from "react-hook-form";
+import logo from "../../../assets/images/PMS 3.png";
+import { Link, useNavigate } from "react-router-dom";
+import { FormData } from "../../../interfaces/Auth";
+import {  useState } from "react";
+import { useToast } from "../../../context/TostifyContext";
+import axios from "axios";
+import {  useToken } from "../../../context/AuthContext";
 
 export default function Login() {
+  // All states here on the top
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+  const { showSuccessToast, showErrorToast } = useToast();
+
+  const {saveAdminData , adminData} = useToken();
+  console.log(adminData)
+  const [spinner, setSpinner] = useState<boolean>(false);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>();
+  const navigate = useNavigate();
+  // test done
+  // Function for password visibility
+  //  void indicates that a function does not return any value
+  const togglePasswordVisibility = (): void => {
+    setShowPassword((prevState: boolean) => !prevState);
+  };
+
+  // senD Data to Api
+  const onSubmit = async (data: FormData) => {
+    setSpinner(true);
+
+    try {
+      const response = await axios.post(
+        "https://upskilling-egypt.com:3003/api/v1/Users/Login",
```

Comment:
I suggest storing the Base URL in a separate file, so that if we want to change it , we don't need to go through all the components to update them.